doc/source/building/working_with_images/custom_fstab_extension.rst

```diff
@@ -0,0 +1,63 @@
+.. _custom_fstab_extension:
+
+Add or Update the Fstab File
+============================
+
+.. sidebar:: Abstract
+
+   This page provides further information for customizing
+   the `/etc/fstab` file which controls the mounting of
+   filesystems at boot time.
+
+In KIWI, all major filesystems that were created at image
+build time are handled by KIWI itself and setup in `etc/fstab`.
+Thus there is usually no need to add entries or change the
+ones added by KIWI. However depending on where the image is
+deployed later it might be required to pre-populate fstab
+entries that are unknown at the time the image is build.
+
+Possible use cases are for example:
+
+* Adding NFS locations that should be mounted at boot time.
+  The information about the NFS location would make this
+  image specific to the target network it later runs in.
+
+* Adding or Changing entries in a read-only root system
+  which becomes effective on first boot but can't be added
+  at that time because of the read-only characteristics.
+
+.. note::
+
+   Modifications to the fstab file are a critical change. If
+   done wrong the risk that the system will not boot exists.
+   In addition this type of modification makes the image
+   specific to its target and creates a dependency to the
+   target hardware, network, etc... Thus this feature should
+   be used with care.
+
+KIWI supports two ways to modify the contents of the `etc/fstab`
+file:
+
+1. Providing an `etc/fstab.append` file
+
+   If that file exists in the image root tree, KIWI will take its
+   contents and append it to the existing `etc/fstab` file. The
+   provided `etc/fstab.append` file will be deleted after successful
+   modification.
+
+2. Providing an `etc/fstab.patch` file
+
+   The `etc/fstab.patch` represents a patch file that will be applied
+   against the `etc/fstab` file using the `patch` program. This method
+   allows also for changing the existing contents of the current
+   `etc/fstab` file. On success the `etc/fstab.patch` will be deleted.
+
+.. note::
+
+   `etc/fstab.append` and `etc/fstab.patch` can be used together.
+   If both files are provided, appending happens first and patch
+   will be last.
+
+The optimal way to provide the fstab extension files is through a
```

Comment:
[suggestion] Move this information up, so that the reader will not start wondering where they should put `/etc/fstab.{patch,append}`

Reply:
will do, good idea